stackrox/stackrox
# ROX-24337: helpful error message to wrong bundle format

## Description

Added Helpful error messages in the case of wrong bundle format being passed through, included cases such as both or no bundles as well.

## Checklist
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed


Manual Testing: I tested for

- [x]  operator bundle

- [x] helm bundle

- [x] both bundles 

- [x] no bundle. 


The following block checks for the validity of the provided init bundle. (`helm install ... -f <init-bundle.yaml>`)
1. Throw an error if no init bundle was provided
2. Throw an error if the wrong init-bundle format was provided (user wrongly passed an operator-formatted init-bundle to helm). We detect the presence of an operator init bundle by checking for the Kubernetes secret annotation "init-bundle.stackrox.io/name", that suggests a user has wrongly used `helm install ... -f <operator-init-bundle.yaml>.`
3. Warn if the user specified both init bundles, e.g. `helm install ... -f <helm-init-bundle.yaml> -f <operator-init-bundle.yaml>`


Compared the expected to the actual result and all 4 checks were correct. 

My input for the command line:

`helm install --dry-run -n stackrox \
   --create-namespace \
   -f /path/to/operator/bundle/file \
   -f /path/to/helm/bundle/file \
   --set central.persistence.none=true \
   --set imagePullSecrets.useExisting=<name> \
   --set clusterName=<name> \
   stackrox-secured-cluster-services \
   `./stackrox-secured-cluster-services-chart``
   
   Depending on the test case, the file paths were removed.
   
   My test case script is: init-bundle.test.yaml where these 4 test cases were checked again.


### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.
